network/nxos/nxos_vxlan_vtep_vni.py

```diff
@@ -0,0 +1,566 @@
+                new_dict[new_key] = value
+            else:
+                new_dict[new_key] = value
+    return new_dict
+
+
+def state_present(module, existing, proposed, candidate):
+    commands = list()
+    proposed_commands = apply_key_map(PARAM_TO_COMMAND_KEYMAP, proposed)
+    existing_commands = apply_key_map(PARAM_TO_COMMAND_KEYMAP, existing)
+
+    for key, value in proposed_commands.iteritems():
+        if key == 'associate-vrf':
+            command = 'member vni {0} {1}'.format(module.params['vni'], key)
+
+            if value:
+                commands.append(command)
+            else:
+                commands.append('no {0}'.format(command))
+
+        elif key == 'peer-ip' and value != 'default':
+            for peer in value:
+                commands.append('{0} {1}'.format(key, peer))
+
+        elif value is True:
+            commands.append(key)
+
+        elif value is False:
+            commands.append('no {0}'.format(key))
+
+        elif value == 'default':
+            if existing_commands.get(key):
+                existing_value = existing_commands.get(key)
+                if key == 'peer-ip':
+                    for peer in existing_value:
+                        commands.append('no {0} {1}'.format(key, peer))
+                else:
+                    commands.append('no {0} {1}'.format(key, existing_value))
+            else:
+                if key.replace(' ', '_').replace('-', '_') in BOOL_PARAMS:
+                    commands.append('no {0}'.format(key.lower()))
+        else:
+            command = '{0} {1}'.format(key, value.lower())
+            commands.append(command)
+
+    if commands:
+        vni_command = 'member vni {0}'.format(module.params['vni'])
+        ingress_replication_command = 'ingress-replication protocol static'
+        interface_command = 'interface {0}'.format(module.params['interface'])
+
+        if ingress_replication_command in commands:
+            static_level_cmds = [cmd for cmd in commands if 'peer' in cmd]
+            parents = [interface_command, vni_command, ingress_replication_command]
+            candidate.add(static_level_cmds, parents=parents)
+            commands = [cmd for cmd in commands if 'peer' not in cmd]
+
+        if vni_command in commands:
+            parents = [interface_command]
+            commands.remove(vni_command)
+            if module.params['assoc_vrf'] is None:
+                parents.append(vni_command)
+            candidate.add(commands, parents=parents)
+
+
+def state_absent(module, existing, proposed, candidate):
+    if existing['assoc_vrf']:
+        commands = ['no member vni {0} associate-vrf'.format(
+                                                module.params['vni'])]
+    else:
+        commands = ['no member vni {0}'.format(module.params['vni'])]
+    parents = ['interface {0}'.format(module.params['interface'])]
+    candidate.add(commands, parents=parents)
+
+
+def main():
+    argument_spec = dict(
```

Comment:
There are more options defined here than in the `DOCUMENTATION:`

Any option specified in `extends_documentation_fragment: nxos` needs defining in `module_utils`, have a look at other network modules.